l10n_th_tax_report/wizard/tax_report_wizard.py

```diff
@@ -0,0 +1,77 @@
+# Copyright 2019 Ecosoft Co., Ltd (https://ecosoft.co.th)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html)
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class TaxReportWizard(models.TransientModel):
+    _name = "tax.report.wizard"
+    _description = "Wizard for Tax Report"
+    # Search Criteria
+    company_id = fields.Many2one(
+        comodel_name="res.company",
+        default=lambda self: self.env.company,
+        string="Company",
+        required=True,
+        ondelete="cascade",
+    )
+    tax_id = fields.Many2one(
+        comodel_name="account.tax",
+        string="Tax",
+        required=True,
+        domain=[
+            ("tax_exigibility", "=", "on_invoice"),
+            ("type_tax_use", "in", ["sale", "purchase"]),
+            ("include_base_amount", "=", False),
+        ],
+    )
+    date_range_id = fields.Many2one(
+        comodel_name="date.range", string="Period", required=True
+    )
+    date_from = fields.Date(string="From Date", required=True)
+    date_to = fields.Date(string="To Date", required=True)
+
+    @api.onchange("date_range_id")
+    def _onchange_date_range_id(self):
+        self.date_from = self.date_range_id.date_start
+        self.date_to = self.date_range_id.date_end
```

Comment:
@kittiu Why we need to onchange this field? why not use "date_start" and "date_end" from "date_range_id" ?

Reply:
Good idea!